exonum/src/encoding/protobuf/mod.rs

```diff
@@ -53,17 +55,17 @@ pub trait ProtobufConvert: Sized {
 
 impl<T> BinaryForm for T
 where
-    T: ProtobufConvert,
-    <T as ProtobufConvert>::ProtoStruct: Message,
+    T: Message,
```

Comment:
I'm too lazy to fetch the PR locally right now, so could you please briefly explain the reason for this change?

Reply:
The idea is that user can use rust-protobuf generated structs or convertible structs(that implement `ProtobufConvert` trait) when serialization is needed.
The main trait that is required to serialize struct in exonum is`Binary Form` but we cannot implement it for T where T is `ProtobufConvert` and for T that is `Message` because this implementations will conflict. So I chose to genearate explicit implementation of this trait for `ProtobufConvert` in `derive(ProtobufConvert)` and implement generic one for `protobuf::Message`

Reply:
I agree that it can create some confusion but I think that we should left it for now.
Suppose we will leave only one way inside and this way will be - use only ProtobufConvertible structs. Then it may happen that some of the original protobuf features are not available through this ProtobufConvert approach(unless some workaround is used), or this ProtobufConvertible structs can be redundant if they do not use types that need to be validated(like transactions with only one numeric field).
* even though I used ProtobufConvert for this trivial transactions to make code change minimal.